quarkchain/evm/specials.py

```diff
@@ -281,6 +284,30 @@ def proc_deploy_root_chain_staking_contract(ext, msg):
     return create_contract(ext, new_msg, target_addr)
 
 
+# 3 inputs: (minter, token ID, amount)
+def proc_mint_mnt(ext, msg):
+    minter = msg.data.extract32(0)
+    token_id = msg.data.extract32(32)
+    amount = msg.data.extract32(64)
+    if amount == 0:
+        return 0, 0, []
+
+    gascost = opcodes.GCALLVALUETRANSFER
+    if not ext.account_exists(minter):
+        gascost += opcodes.GCALLNEWACCOUNT
+
+    if msg.gas < gascost:
+        return 0, 0, []
+
+    allowed_sender, _ = _system_contracts[SystemContract.NON_RESERVED_NATIVE_TOKEN]
+    # Only system contract has access to minting new token
+    if allowed_sender != msg.sender:
+        return 0, 0, []
+
+    ext._state.delta_token_balance(minter, token_id, amount)
+    return 1, msg.gas - gascost, [0] * 31 + [1]
```

Comment:
@ninjaahhh @hanyunx why return [0] * 31 + [1] here?

Reply:
to indicate that token is successfully minted.